### PR TITLE
Save manager improvements

### PR DESCRIPTION
**Pull request by flash-fire**
Make size only 60% height of window screen initially.
Add ability to delete/select multiple rows.
Add ability to open the directory of the save.

### CODE REVIEW

**Comment by Asinin3** [comment]
These improvements are really great, everything is working for me too didn't notice any problems.

File: rpcs3/rpcs3qt/save_manager_dialog.cpp [hunk 1]
```diff
@@ -249,36 +256,118 @@ void save_manager_dialog::OnEntryRemove()
...
-	//Events
-	connect(removeAct, &QAction::triggered, this, &save_manager_dialog::OnEntryRemove);
+	// Events
+	connect(removeAct, &QAction::triggered, this, &save_manager_dialog::OnEntriesRemove); // entriesremove handles case of one as well
 	connect(infoAct, &QAction::triggered, this, &save_manager_dialog::OnEntryInfo);
+	connect(showDirAct, &QAction::triggered, [=]() {
+		int idx_real = m_list->item(idx, 0)->data(Qt::UserRole).toInt();
+		QString path = qstr(m_dir + m_save_entries[idx_real].dirName + "/");
+
+		QProcess* process = new QProcess();
```
Comment (danilaml) [inline rpcs3/rpcs3qt/save_manager_dialog.cpp]: This leaks. You can just use `QProcess::startDetached(command, QStringList(params))` instead. Also, if you only want to show specific dir (without highlighting specific file) [openURL](http://doc.qt.io/qt-5.9/qdesktopservices.html#openUrl) should work in a cross-platform way.

Comment (flash-fire) [reply rpcs3/rpcs3qt/save_manager_dialog.cpp]: Thanks. I fixed the other use of improper use of QProcess as well with this information.

File: rpcs3/rpcs3qt/save_manager_dialog.cpp [hunk 2]
```diff
@@ -249,36 +257,107 @@ void save_manager_dialog::OnEntryRemove()
 	}
 }
 
+void save_manager_dialog::OnEntriesRemove()
+{
+	QList<QTableWidgetItem*> items = m_list->selectedItems();
```
Comment (Megamouse) [inline rpcs3/rpcs3qt/save_manager_dialog.cpp]: why not use selectedRows?

File: rpcs3/rpcs3qt/save_manager_dialog.cpp [hunk 3]
```diff
@@ -249,36 +257,107 @@ void save_manager_dialog::OnEntryRemove()
...
+		{
+			fs::remove_all(m_dir + m_save_entries[idx_real].dirName + "/");
+		}
+	}
+}
+
 //Pop-up a small context-menu, being a replacement for save_data_manage_dialog
 void save_manager_dialog::ShowContextMenu(const QPoint &pos)
 {
+	bool selectedItems = m_list->selectedItems().size() > 4; // > 4 is slight hack due to four columns-- more than one row selected.
```
Comment (Megamouse) [inline rpcs3/rpcs3qt/save_manager_dialog.cpp]: same here

File: rpcs3/rpcs3qt/save_manager_dialog.cpp [hunk 4]
```diff
@@ -249,36 +257,80 @@ void save_manager_dialog::OnEntryRemove()
...
 
+void save_manager_dialog::OnEntriesRemove()
+{
+	QModelIndexList selection(m_list->selectionModel()->selectedRows());
+	if (selection.size() == 0)
+	{
+		return;
+	}
+
+	// Small hack. Divide by four because of columns.
```
Comment (Megamouse) [inline rpcs3/rpcs3qt/save_manager_dialog.cpp]: lol

**Comment by Marthgun** [comment]
can u add a check if user is fullscreen, then exit fullscreen automatically to open up the save/load dialog?  being in fullscreen mode will open freeze game, open dialog behind fullscreen.  which is not great to navigate to.

also, any way to poll the dialog for gamepad input?  or set default generic directional and or accept/cancel buttons?  idk if u want to avoid hacky stuff like this but it is quite annoying not being able to use gamepad

**Comment by Megamouse** [comment]
sadly nothing you said is easily done and is already known to the devs anyway

File: rpcs3/rpcs3qt/save_manager_dialog.cpp (hunk 4 above)
Comment (flash-fire) [reply rpcs3/rpcs3qt/save_manager_dialog.cpp]: lol

**Pull request merged**